enum.js

```diff
@@ -19,7 +19,7 @@ class Enum {
     }
     for (const c of Object.getOwnPropertyNames(cases)) {
       if (Object.is(this.value, c)) {
-        return cases[c]();
+        return cases[c](this);
```

Comment:
This change, which is tested [here](https://github.com/Thrillberg/imperial/pull/15/files#diff-3c42b63f302a4effa1f7f78aa92faa94R75-R82), is for now just a convenience to let us use functions like [these](https://github.com/Thrillberg/imperial/pull/15/files#diff-fcd5628e27bdebc81efd345a905292c3R23-R25) in the branches of each `when`. Although it's just for convenience now, later it will become the foundation of a "tagged union" construct, the details of which are pretty fuzzy to me.